### PR TITLE
Add more functional tests individual bmh selection

### PR DESCRIPTION
**Pull request by rabi**

### CODE REVIEW

File: tests/functional/base_test.go [hunk 1]
```diff
@@ -132,6 +132,54 @@ func TwoNodeBaremetalSetSpecWithWrongNodeLabel(namespace string) map[string]inte
...
 
+func MultiNodeBaremetalSetSpecWithSameNodeLabel(namespace string) map[string]interface{} {
+	spec := map[string]interface{}{
+		"baremetalHosts": map[string]interface{}{
+			"compute-0": map[string]interface{}{
+				"ctlPlaneIP":       "10.0.0.1",
+				"bmhLabelSelector": map[string]string{"nodeType": "compute"},
+			},
+			"compute-1": map[string]interface{}{
+				"ctlPlaneIP":       "10.0.0.1",
```
Comment (abays) [inline tests/functional/base_test.go:143]: Should we use different IPs between the hosts to have more parity with real CRs?

Comment (rabi) [reply tests/functional/base_test.go]: done! they're not relevant in the webhook tests though.

Comment (abays) [reply tests/functional/base_test.go]: Fair point in that regard, but thanks for updating nonetheless :+1:

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**